src/main/java/carbon/TaskList.java

```diff
@@ -0,0 +1,234 @@
+import carbon.task.Deadline;
+import carbon.task.Event;
+import carbon.task.Task;
+import carbon.task.Todo;
+
+/**
+ * Contains and manages user-declared tasks.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Constructs an instance of TaskList class.
+     *
+     * @return TaskList object.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Loads and adds a task to the list of all tasks.
+     *
+     * @param data String input of the encoded task.
+     * @throws CarbonException  If an error is encountered while decoding the String data.
+     */
+    public void loadTask(String data) throws CarbonException {
+        try {
+            Task task = Task.decodeTask(data);
+            this.tasks.add(task);
+        } catch (CarbonException error) {
+            throw error;
+        }
+    }
+
+    /**
+     * Encodes the task into text data to be stored.
+     * Does not use any actual encoding format, but simplifies the text.
+     *
+     * @return Encoded task data.
+     */
+    public String encodeTasks() {
+        String encodedTasks = "";
+        int len = this.tasks.size();
+        for (int i = 0; i < len; i++) {
+            encodedTasks += this.tasks.get(i).encode();
+        }
+        return encodedTasks;
+    }
+
+    /**
+     * Updates if a specific task is done or not.
+     * Performs input validation before executing.
+     *
+     * @param input User text input.
+     * @param isDone Whether the task is done or not.
+     * @return Execution log.
+     * @throws CarbonException  If there are invalid parameters.
+     */
+    public String validateAndMark(String input, boolean isDone) throws CarbonException {
+        int taskNumber;
+        int len = input.length();
+        if (isDone) {
+            int requiredLen = "mark ".length();
+            if (len <= requiredLen) {
+                CarbonException invalidParam = new InvalidParamException(input);
+                throw invalidParam;
+            } else {
+                taskNumber = Integer.valueOf(input.substring(requiredLen));
+            }
+        } else {
+            int requiredLen = "unmark ".length();
+            if (len <= requiredLen) {
+                CarbonException invalidParam = new InvalidParamException(input);
+                throw invalidParam;
+            } else {
+                taskNumber = Integer.valueOf(input.substring(requiredLen));
+            }
+        }
+
+        if (taskNumber < 1 || taskNumber > this.tasks.size()) {
+            CarbonException outOfBounds = new OutOfBoundsException(taskNumber, this.tasks.size());
+            throw outOfBounds;
+        } else {
+            String log = this.setTaskDoneness(taskNumber, isDone);
+            return log;
+        }
+    }
+
+    private String setTaskDoneness(int taskNumber, boolean isDone) {
+        Task task = this.tasks.get(taskNumber - 1);
+        task.changeDoneness(isDone);
```

Comment:
changeDoneness can also have a formal name like changeCompleteStatus

Reply:
Actually shouldn't this probably follow the same format as above? Something like `setDone`?